lib/orbit/evented.js

```diff
@@ -191,72 +191,36 @@ var Evented = {
       return listeners;
     },
 
-    resolve: function(eventNames) {
-      var listeners = this.listeners(eventNames),
-          args = Array.prototype.slice.call(arguments, 1);
-
-      return new Orbit.Promise(function(resolve, reject) {
-        var resolveEach = function() {
-          if (listeners.length === 0) {
-            reject();
-          } else {
-            var listener = listeners.shift();
-            var response = listener[0].apply(listener[1], args);
-
-            if (response) {
-              response.then(
-                function(success) {
-                  resolve(success);
-                },
-                function(error) {
-                  resolveEach();
-                }
-              );
-            } else {
-              resolveEach();
-            }
-          }
-        };
-
-        resolveEach();
-      });
+    resolve(eventNames, ...args) {
```

Comment:
Hmmm ... I agree that this should be changed, but probably not with this PR. It would be better to get this branch green first and submit a separate PR to change the name everywhere. 

I haven't seen a comparable method in any other promise lib, so we should probably give this a unique name (not `resolve`).

> I'm pretty sure Promise.all allows parallel execution

That's correct.